### ISSUE TITLE
Created a one-hit special with an improper combination

### ISSUE THREAD

**Issue by nvs** [nvs/gem#229]
It happened in [this game](http://storage.entgaming.net/replay/view_replay.php?file=8361506.w3g). I was offered a Perfect Diamond, Flawless Diamond, and Normal Aquamarine. I was given the option to one-hit and then created a Black Opal.

I went too far into the replay and need to start over again. But I need to check the following:
- What my previous round selection was;
- If anyone else was placing or selecting gems at the same time.

My hunch says another bug related to waits and certain variables going out of scope. But perhaps it was something I changed? I cannot remember anything right now. But I'll look into it.

**Comment by nvs** [comment]
Okay, I'm pretty sure I created this when dealing with #173 (but I am not entirely certain). Apparently, I overthought the issue again. I think in my previous testing I realized that the units were not `null`, but they also weren't anything else. However, I failed to recognize what was plainly in front of me, which is that they return `0` when passed to `GetUnitTypeId ()`. So, a much smaller fix for the same issue in #173.

Anyways, all of this becomes less of an issue when #168 is finished in 1.4.
